worlds/yugioh06/Client_bh.py

```diff
@@ -0,0 +1,134 @@
+
+    async def game_watcher(self, ctx: BizHawkClientContext) -> None:
+        try:
+            read_state = await bizhawk.read(ctx.bizhawk_ctx,
+                                            [(0x0, 8, "EWRAM"),
+                                            (0x52e8, 32, "EWRAM"),
+                                            (0x5308, 32, "EWRAM"),
+                                            (0x5325, 1, "EWRAM"),
+                                            (0x6c38, 4, "EWRAM")])
+            game_state = bytes([byte for byte in read_state[0] if byte != 0]).decode("utf-8")
+            locations = read_state[1]
+            items = read_state[2]
+            amount_items = int.from_bytes(read_state[3], "little")
+            money = int.from_bytes(read_state[4], "little")
+
+            # make sure save was created
+            if game_state != 'YWCT2006':
+                return
+            await bizhawk.write(ctx.bizhawk_ctx, [(0x5308, parse_items(bytearray(items), ctx.items_received), "EWRAM")])
+            money_received = 0
+            for item in ctx.items_received:
+                if item.item == item_to_index["5000DP"] + 5730000:
+                    money_received += 1
+            if money_received > amount_items:
+                await bizhawk.write(ctx.bizhawk_ctx,
+                                [(0x6c38, (money + (money_received - amount_items) * 5000).to_bytes(4, "little"),
+                                  "EWRAM"),
+                                 (0x5325, money_received.to_bytes(4, "little"), "EWRAM")])
+
+            locs_to_send = set()
+
+            # Check for set location flags.
+            for byte_i, byte in enumerate(bytearray(locations)):
+                for i in range(8):
+                    and_value = 1 << i
+                    if byte & and_value != 0:
+                        flag_id = byte_i * 8 + i
+
+                        location_id = flag_id + 5730001
+                        if location_id in ctx.server_locations:
+                            locs_to_send.add(location_id)
+
+            # Send locations if there are any to send.
+            if locs_to_send != self.local_checked_locations:
+                self.local_checked_locations = locs_to_send
+
+                if locs_to_send is not None:
+                    await ctx.send_msgs([{
+                        "cmd": "LocationChecks",
+                        "locations": list(locs_to_send)
+                    }])
+
+                # Send game clear if we're in either any ending cutscene or the credits state.
+            if not ctx.finished_game and bytearray(locations)[18] & (1 << 5) != 0:
+                await ctx.send_msgs([{
+                    "cmd": "StatusUpdate",
+                    "status": ClientStatus.CLIENT_GOAL
+                }])
+
+        except bizhawk.RequestFailedError:
+            # Exit handler and return to main loop to reconnect.
+            pass
+
+
+def parse_items(localItems, items: List[NetworkItem]):
```

Comment:
Would be nice to have a docstring for this function and a typehint for `localItems` (which should probably be `local_items` for consistency).